### PR TITLE
feat: Deploy with automine, but wait for blocktime to catch up

### PR DESCRIPTION
**Pull request by phated**
This changes the hardhat node defaults to automine and no interval mine. This speeds up deployments; however, it accelerates block time by ~20 seconds ahead of the wall clock. Since the DF client uses real-world time (wall clock) to do voyage estimations and stuff, we need to wait until the wall clock catches up to the chain. Once the time syncs, we switch on interval mining to have an accurate client experience.

This drops deploy time from ~90 seconds to ~30 seconds (even with the ~20 second wait time).

### CODE REVIEW

File: eth/test/utils/TestWorld.ts [hunk 1]
```diff
@@ -63,11 +63,6 @@ export async function initializeWorld({
 }: InitializeWorldArgs): Promise<World> {
   const [deployer, user1, user2] = await hre.ethers.getSigners();
 
-  // The tests assume that things get mined right away
-  // TODO(#912): This means the tests are wildly fragile and probably need to be rewritten
```
Comment (cha0sg0d) [inline eth/test/utils/TestWorld.ts:67]: glad we're getting rid of this code

**Review by cha0sg0d — COMMENTED** [review]
Game starts as expected.

I'd like to add a script to start node+contracts, client, and hot reloading all at once. Is there a tool you recommend?

File: eth/test/utils/TestWorld.ts (hunk 1 above)
Comment (phated) [reply eth/test/utils/TestWorld.ts:67]: I removed these, but the **issue** doesn't go away. Ideally, we'd get hardhat to support interval mining within it's test helpers. I'm going to talk to Franco

**Comment by phated** [comment]
> I'd like to add a script to start node+contracts, client, and hot reloading all at once.

We already use `npm-run-all` for this stuff. Ideally this would all move into hardhat's `node:ready` task, like I did in [afterimage](https://github.com/phated/afterimage/blob/main/tasks/node.ts) but the project's structure doesn't allow for that.

**Pull request merged**